presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -6630,4 +6631,56 @@ public void testInnerJoinWithOrCondition()
         sql = "select o.orderkey, l.partkey, l.suppkey from lineitem l join orders o on (l.orderkey = o.orderkey or (l.partkey + l.suppkey) = o.orderkey) and l.quantity*totalprice > 1000 where l.quantity < 5 and o.totalprice < 2000";
         assertQuery(enablePushFilterOptimization, sql);
     }
+
+    @Test
+    public void testPreProcessMetastoreCalls()
```

Comment:
can you add tests for error handling (e.g. table/column/view doesn't exist)?

Reply:
Added for table/column/catalog/schema.